File: OrangeJuice.Server.Core.FSharp/Filters/PrimaryVariantlItemFilter.fs
namespace OrangeJuice.Server.FSharp.Services

open System.Xml
open System.Xml.Linq
open System.Xml.XPath

open OrangeJuice.Server.Filters

type PrimaryVariantlItemFilter() =
    interface IFilter<XElement> with
        member this.Filter(element : XElement) : bool =
            let nm = new XmlNamespaceManager(new NameTable())
            nm.AddNamespace("x", element.Name.Namespace.ToString()) |> ignore

            Seq.isEmpty(element.XPathSelectElements("x:ImageSets/x:ImageSet[contains(@Category, 'variant')]", nm))

File: OrangeJuice.Server.Core/Services/PrimaryVariantlItemFilter.cs
using System.Linq;
using System.Xml;
using System.Xml.Linq;
using System.Xml.XPath;

using OrangeJuice.Server.Filters;

namespace OrangeJuice.Server.Services
{
	public sealed class PrimaryVariantlItemFilter : IFilter<XElement>
	{
		public bool Filter(XElement element)
		{
			var nm = new XmlNamespaceManager(new NameTable());
			nm.AddNamespace("x", element.Name.Namespace.ToString());

			return !element.XPathSelectElements("x:ImageSets/x:ImageSet[contains(@Category, 'variant')]", nm).Any();
		}
	}
}